charger/daheimladen.go

```diff
@@ -0,0 +1,243 @@
+package charger
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/charger/daheimladen"
+	"github.com/evcc-io/evcc/util"
+	"github.com/evcc-io/evcc/util/request"
+)
+
+// DaheimLaden charger implementation
+type DaheimLaden struct {
+	*request.Helper
+	chargingStationID string
+	connectorID       int32
+	idTag             string
+	token             string
+	transactionID     int32
+}
+
+func init() {
+	registry.Add("daheimladen", NewDaheimLadenFromConfig)
+}
+
+// NewDaheimLadenFromConfig creates a DaheimLaden charger from generic config
+func NewDaheimLadenFromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := struct {
+		Token             string
+		ChargingStationID string
+		IDTag             string
+	}{}
+
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	return NewDaheimLaden(cc.Token, cc.ChargingStationID, cc.IDTag)
+}
+
+// NewDaheimLaden creates DaheimLaden charger
+func NewDaheimLaden(token string, chargingStationID string, idTag string) (*DaheimLaden, error) {
+	c := &DaheimLaden{
+		Helper:            request.NewHelper(util.NewLogger("daheim")),
+		chargingStationID: chargingStationID,
+		connectorID:       1,
+		idTag:             idTag,
+		token:             token,
+	}
+	return c, nil
+}
+
+// Enabled implements the api.Charger interface
+func (c *DaheimLaden) Enabled() (bool, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("https://api.daheimladen.com/v1/cs/%s/status", c.chargingStationID), nil)
+	if err != nil {
+		return false, err
+	}
+	statusRes := daheimladen.GetLatestStatus{}
+	err = c.DoJSON(req, statusRes)
+	if err != nil {
+		return false, err
+	}
+
+	if statusRes.Status == "Preparing" {
```

Comment:
What would be the value during charging? Also still "Preparing"? If not, please add these values.